### PR TITLE
feat(sdk): finetune decorator supports auto build and validation datasets

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
- add `--val-dataset` argument for `swcli model run`, env is `SW_FINETUNE_VALIDATION_DATASET_URI`   @anda-ren 
- add auto model build  for finetune decorator

## Modules
- [x] Client
- [x] Python-SDK

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/consts/env.py [hunk 1]
```diff
@@ -8,6 +8,7 @@ class SWEnv:
     status_dir = "SW_TASK_STATUS_DIR"
     log_dir = "SW_TASK_LOG_DIR"
     dataset_uri = "SW_DATASET_URI"
+    finetune_validation_dataset_uri = "SW_FINETUNE_VALIDATION_DATASET_URI"
```
Comment (goldenxinxing) [inline client/starwhale/consts/env.py:11]: The server env named as 'SW_VALIDATION_DATASET_URI', so, keep consistent with server? cc @anda-ren

Comment (anda-ren) [reply client/starwhale/consts/env.py:11]: It's fine, I could modify it in the server side

Comment (goldenxinxing) [reply client/starwhale/consts/env.py:11]: ok

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**